include/llbuild/Basic/Utility.h

```diff
@@ -0,0 +1,35 @@
+//===- Utility.h ------------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_BASIC_UTILITY_H
+#define LLBUILD_BASIC_UTILITY_H
+
+#include "llvm/ADT/StringRef.h"
+#include <string>
+
+using namespace llvm;
+
+namespace llbuild {
+namespace basic {
+
+/// Escapes control characters in a string to make them displayable on terminal.
+/// For e.g. hello\nworld -> hello\\nworld, input A -> input\ A
+///
+/// \param argument Reference of string to be escaped.
+///
+/// \returns Escaped string.
+std::string shellEscaped(StringRef argument);
```

Comment:
We should make this render into a `SmallVector<char>` instead of returning an `std::string`. This lets the client avoid heap allocation if desired, and is a common pattern in LLVM/Clang/Swift. `FileSystem.h` has examples of the pattern I am talking about.